05/app.js

```diff
@@ -0,0 +1,54 @@
+function Student(firstName, lastName) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.grades = {}
+}
+
+Student.prototype.addGrade = function (subject, grade) {
+    if(typeof this.grades[subject] === 'undefined') {
+        this.grades[subject] = [];
+    }
+    
+    this.grades[subject].push(grade);
+}
+
+Student.prototype.getAverageGrade = function (subject) {
+    
+    if(subject !== undefined) {
```

Comment:
Raz robisz prawidłowo, raz nie.
Zapamiętaj, że jak sprawdzasz typ to użyj `typeof ` :)